pkg/controllers/vpcnetworkconfig/vpcnetworkconfiguration_controller.go

```diff
@@ -0,0 +1,94 @@
+/* Copyright © 2022 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package vpcnetworkconfig
+
+import (
+	"context"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	v1alpha1 "github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+)
+
+// VPCNetworkConfigurationReconciler reconciles a VPCNetworkConfiguration object.
+type VPCNetworkConfigurationReconciler struct {
+	client.Client
+	Scheme                     *runtime.Scheme
+	installedVPCNetworkConfigs *VPCNetworkConfigInfoCache
+}
+
+func NewVPCNetworkConfigurationReconciler(client client.Client, scheme *runtime.Scheme) *VPCNetworkConfigurationReconciler {
+	return &VPCNetworkConfigurationReconciler{
+		Client:                     client,
+		Scheme:                     scheme,
+		installedVPCNetworkConfigs: NewVPCNetworkConfigInfoStore(),
+	}
+}
+
+// Reconcile will attempt to ensure that the imported Resource is installed in local cluster as per the
+// VPCNetworkConfiguration object.
+func (r *VPCNetworkConfigurationReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	klog.V(2).InfoS("Reconciling VPCNetworkConfiguration", "vpcNetworkConfiguration", req.Name)
+
+	var vpcNetworkConfig v1alpha1.VPCNetworkConfiguration
+	err := r.Client.Get(ctx, req.NamespacedName, &vpcNetworkConfig)
+	var isDeleted bool
+	if err != nil {
+		isDeleted = apierrors.IsNotFound(err)
+		if !isDeleted {
+			klog.InfoS("Unable to fetch VPCNetworkConfiguration", "vpcNetworkConfiguration", req.NamespacedName.String(), "err", err)
+			return ctrl.Result{}, err
+		} else {
+			vpcNetworkConfigurationObj, exist := r.installedVPCNetworkConfigs.GetByKey(req.NamespacedName.String())
```

Comment:
logic here is different with other controller. The other controller will return error if it is not IsNotFound(err). Any reason here?